elide-core/src/main/java/com/yahoo/elide/graphql/ModelBuilder.java

```diff
@@ -168,27 +169,57 @@ public GraphQLSchema build() {
                     .argument(pageFirstArgument)
                     .argument(pageOffsetArgument)
                     .argument(buildInputObjectArgument(clazz, true))
-                    .type(new GraphQLList(buildQueryObject(clazz))));
+                    .type(buildConnectionObject(clazz)));
         }
 
         /*
          * Walk the object graph (avoiding cycles) and construct the GraphQL output object types.
          */
-        dictionary.walkEntityGraph(rootClasses, this::buildQueryObject);
+        dictionary.walkEntityGraph(rootClasses, this::buildConnectionObject);
 
         /* Construct the schema */
         GraphQLSchema schema = GraphQLSchema.newSchema()
                 .query(root)
                 .mutation(root)
                 .build(new HashSet<>(CollectionUtils.union(
-                        queryObjectRegistry.values(),
+                        connectionObjectRegistry.values(),
```

Comment:
I'm not sure this is right.  The registry is meant to be a container for types.  It should include the query Objects

Reply:
Connection objects retain the top-level name for types. i.e. `book` or `author`. This registry is required for GraphQLTypeReferences. I don't think we would ever reference a `__node__author`.

Reply:
👍